### DISCUSSION TITLE
Why is .clangd file not used?

### DISCUSSION THREAD

**Discussion by juxeii** [clangd/clangd#1679] [Q&A] [answered]
Hi,

I have created a .`clangd `file at project root.
Then I start `clangd `from the project root:

```
clangd --log=verbose
clangd is a language server that provides IDE-like features to editors.

It should be used via an editor plugin rather than invoked directly. For more information, see:
        https://clangd.llvm.org/
        https://microsoft.github.io/language-server-protocol/

clangd accepts flags on the commandline, and in the CLANGD_FLAGS environment variable.

I[22:08:28.036] clangd version 15.0.7
I[22:08:28.036] Features: linux+grpc
I[22:08:28.036] PID: 5994
I[22:08:28.036] Working directory: /var/fpwork/reiss/gnb/cplane
I[22:08:28.036] argv[0]: clangd
I[22:08:28.036] argv[1]: --log=verbose
V[22:08:28.036] User config file is /home/reiss/.config/clangd/config.yaml
I[22:08:28.036] Starting LSP over stdin/stdout
```
How can one tell if the .`clangd` file is used?
Even if I introduce syntax erros in the .`clangd `file, nothing changes.
.`clangd `just looks like this:
```
CompileFlags:
  CompilationDatabase: ./build/compile_commands.json
```
`Clangd `looks for a file `/home/reiss/.config/clangd/config.yaml`, but this does not exist? Does it have to?

**Comment by sam-mccall** [comment, marked as answer]
You didn't open any files yet.
Each file is governed by config files in ancestor directories.

**Comment by juxeii** [comment]
Thx, makes sense. I thought that the working directory, from which `clangd `has been started, is also scanned/reported for a `.clangd` file.
